errors/ControllerErrorHandler.ts

```diff
@@ -0,0 +1,72 @@
+import 'reflect-metadata';
+import {Request, Response} from 'express';
+import httpStatus = require('http-status');
+
+/**
+ * Default error handler. Does console.error with the error and sets the
+ * http response code to 500.
+ *
+ * @param {Request} req - express request object
+ * @param {Response} res - express response object
+ * @param {Error} err - error that happend
+ */
+export const DEFAULT_ERROR_HANDLER = (req, res, err) => {
+    console.error(err);
+    res.status(httpStatus.INTERNAL_SERVER_ERROR).end();
+};
+
+/**
+ * @typedef ErrorHandlerFunction
+ * @type {Function}
+ * @param {Request} request - ExpressJS request object.
+ * @param {Response} response - ExpressJS response object.
+ * @param {Error} error - The error that happend.
+ */
+export type ErrorHandlerFunction<T extends Error> = (request: Request, response: Response, error: T) => void;
+
+/**
+ * Manager object that handles all ErrorHandlerFunctions for a controller. Registers itself in the controllers
+ * metadata. If an error is thrown, just pass it to the error handler. The handler will do all the magic.
+ *
+ * @class
+ */
+export class ControllerErrorHandler {
```

Comment:
It's the errorhandler that is attached to the controller and since `ErrorHandler` is already exposed as the decorator for the method, that name was already taken :)